src/utils.zig

```diff
@@ -1919,6 +1919,16 @@ pub const BinaryReader = struct {
 		return std.mem.readInt(T, self.remaining[0..bufSize], self.endian);
 	}
 
+	pub fn readFloat(self: *BinaryReader, comptime T: type) error{OutOfBounds, IntOutOfBounds}!T {
```

Comment:
I made this comptime so switch is optimized away, since data layouts are usually compile time defined it should not be a problem.

Reply:
What exactly did you make comptime? I don't see any comptime that isn't required by the language.

Reply:
Compare `readInt` and `readFloat` signatures:

```zig
pub fn readInt(self: *BinaryReader, T: type) error{OutOfBounds, IntOutOfBounds}!T
```
```zig
pub fn readFloat(self: *BinaryReader, comptime T: type) error{OutOfBounds, IntOutOfBounds}!T
```

Maybe it is required and I just don't know it.

Reply:
Ok